solidity/truffle-examples/random-datasource/test/utils.js

```diff
@@ -0,0 +1,25 @@
+const fs = require('fs')
+const path = require('path')
+const API_PATH = path.resolve(__dirname, '../apikeys.js')
+
+const PREFIX = 'Returned error: VM Exception while processing transaction: '
+
+const waitForEvent = (_event, _from = 0, _to = 'latest') =>
+  new Promise ((resolve, reject) =>
+    _event({ fromBlock: _from, toBlock: _to }, (e, ev) =>
+      e ? reject(e) : resolve(ev)))
+
+const fileExists = _path => fs.existsSync(_path)
+
+const getInfuraKey = () => fileExists(API_PATH)
```

Comment:
This is still a copy/paste.

Reply:
Changed that function to the more general getExternalVariable, and exported where needed.